galsim/chromatic.py

```diff
@@ -131,6 +136,216 @@ def __init__(self, obj):
+        """
+        A routine to force an object for which interpolation was previously set up go back to the
+        exact calculation.
+        """
+        if hasattr(self, 'waves'):
+            # Check whether some chromatic transformation was done when setting up the
+            # interpolation, and restore the associated _A and _fluxFactor in combination with any
+            # other transformations that have been done in the meantime.
+            if hasattr(self, '_save_A'):
+                # Need to be careful about order.  What's in self._A came after what's in
+                # self._save_A.
+
+                if all([self._nullTransformation(w) for w in self.waves]):
+                    self._A = self._save_A
+                else:
+                    tmp_A = self._A
+                    self._A = lambda w: tmp_A(w) * self._save_A
+
+                if all([self._nullFluxTransformation(w) for w in self.waves]):
+                    self._fluxFactor = self._save_fluxFactor
+                else:
+                    tmp_fluxFactor = self._fluxFactor
+                    self._fluxFactor = lambda w: tmp_fluxFactor(w) * self._save_fluxFactor(w)
+
+                # Then delete the old _save_A.
+                del self._save_A
+                del self._save_fluxFactor
+
+            # Get rid of the stored attributes related to interpolation.
+            del self.waves
+            del self.stepK_vals
+            del self.maxK_vals
+            del self.ims
+
+    def hasInterpolation(self):
+        """
+        A routine to check whether an object has interpolation set up or not.
+
+        @returns    True or False
+        """
+        return hasattr(self, 'waves')
+
+    def _evaluateAtWavelength(self, wave):
+        """
+        Evaluate this ChromaticObject at a particular wavelength, either using interpolation or via
+        direct calculation, depending on whether or not interpolation has been set up.
+
+        @param wave     Wavelength in nanometers.
+
+        @returns the monochromatic object at the given wavelength, as a GSObject.
+        """
+        if hasattr(self, 'waves'):
+            im, stepk, maxk = self._imageAtWavelength(wave)
+            return galsim.InterpolatedImage(im, _force_stepk=stepk, _force_maxk=maxk)
+        else:
+            return self.evaluateAtWavelength(wave)
+
+    def _imageAtWavelength(self, wave):
+        """
+        Get an image of the object at a particular wavelength, using linear interpolation between
+        the originally-stored images.  Also returns values for step_k and max_k, to be used to
+        expedite the instantation of InterpolatedImages.
+
+        @param wave     Wavelength in nanometers.
+
+        @returns an Image of the object at the given wavelength.
+        """
+        # First, some wavelength-related sanity checks.
+        if not hasattr(self, 'waves'):
+            raise RuntimeError("Requested image at some wavelength without setting up "
```

Comment:
Shouldn't have these checks in private methods.  Checks like this should always be in the top-level public method, not in private methods, since if the code is written correctly, they should never hit.

Sometimes it is useful to put them in when developing code to help catch mistakes, but at this point, they should be removed as they just slow down the code.